### PR TITLE
Hosts grid

### PR DESCRIPTION
**Pull request by mireaulf**
Closes https://github.com/preludeorg/build/issues/490

image

- Not filtering data based on status
- Download not working
- List is not working
- Does not sort

No logic, only a placeholder for data

### CODE REVIEW

**Comment by mireaulf** [comment]
Good enough for PR?

File: apps/detect/src/components/hosts-grid/hosts-grid.module.css [hunk 1]
```diff
@@ -0,0 +1,149 @@
...
+    font-size: 14px;
+    line-height: 20px;
+}
+
+.gridActions {
+    display: flex;
+    flex-direction: row;
+    align-items: center;
+    padding: 0;
+    gap: 16px;
```
Comment (stephanwampouille) [inline apps/detect/src/components/hosts-grid/hosts-grid.module.css:28]: I think you're missing a margin of 1rem for the bottom of this

File: apps/detect/src/components/hosts-grid/hosts-grid.module.css [hunk 2]
```diff
@@ -0,0 +1,149 @@
...
+.statuses div {
+    display: flex;
+    flex-direction: row;
+    align-items: center;
+    padding: 0;
+    gap: 4px;
+}
+
+.statusError {
+    fill: #ffbf44;
```
Comment (stephanwampouille) [inline apps/detect/src/components/hosts-grid/hosts-grid.module.css:114]: not fully necessary but we have css variables for all of the fills and backgrounds if you want to switch to them or I can also help and do that

File: apps/detect/src/components/hosts-grid/hosts-grid.tsx [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+
+    const hostsInError = hosts.filter(h => h.status == "error")
+    const hostsUnprotected = hosts.filter(h => h.status == "unprotected")
+    const hostsProtected = hosts.filter(h => h.status == "protected")
+    return <div className={styles.viz}>
+        <div className={styles.stats}>
+            <div className={styles.statsHosts}>{hosts.length} hosts</div>
+            <div className={styles.gridActions}>
+                <div className={styles.gridAction}>
+                    List
```
Comment (stephanwampouille) [inline apps/detect/src/components/hosts-grid/hosts-grid.tsx]: missing list icon

File: apps/detect/src/components/hosts-grid/hosts-grid.tsx [hunk 4]
```diff
@@ -0,0 +1,58 @@
...
+    /* -- */
+
+    const hostsInError = hosts.filter(h => h.status == "error")
+    const hostsUnprotected = hosts.filter(h => h.status == "unprotected")
+    const hostsProtected = hosts.filter(h => h.status == "protected")
+    return <div className={styles.viz}>
+        <div className={styles.stats}>
+            <div className={styles.statsHosts}>{hosts.length} hosts</div>
+            <div className={styles.gridActions}>
+                <div className={styles.gridAction}>
```
Comment (stephanwampouille) [inline apps/detect/src/components/hosts-grid/hosts-grid.tsx]: You can also replace these with the <Button> component that we created but not fully necessary. I can help make the switch

**Review by stephanwampouille — CHANGES_REQUESTED** [review]

**Review by kenrick — APPROVED** [review]

**Pull request merged**